hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/groupschart/client/GroupsPieChartWidget.java

```diff
@@ -0,0 +1,205 @@
+            }
+            unassignedTargets = totalTargetCount - sum;
+        }
+
+        draw();
+
+    }
+
+    private static PieArc getPie(Long count, Long total, double startAngle) {
+        final Double percentage = count.doubleValue() / total.doubleValue();
+        return new PieArc(startAngle, startAngle + percentage * 2 * Math.PI);
+    }
+
+    private void draw() {
+        if (svg == null || groupTargetCounts == null || totalTargetCount == null) {
+            return;
+        }
+
+        final Array<Arc> dataArray = Array.create();
+
+        PieArc pie = getPie(unassignedTargets, totalTargetCount, 0);
+        dataArray.push(pie.getArc());
+
+        double lastAngle = pie.getEndAngle();
+        for (int i = 0; i < groupTargetCounts.size(); i++) {
+            final PieArc arcEntry = getPie(groupTargetCounts.get(i), totalTargetCount, lastAngle);
+            dataArray.push(arcEntry.getArc());
+            lastAngle = arcEntry.getEndAngle();
+        }
+
+        UpdateSelection pies = pieGroup.selectAll(".pie").data(dataArray);
+        pies.enter().append("path").classed("pie", true).on(BrowserEvents.MOUSEOVER, new DatumFunction<Void>() {
+            @Override
+            public Void apply(Element context, Value d, int index) {
+                Array<Double> point = arc.centroid(d.as(Arc.class), index);
+                double x = point.getNumber(0);
+                double y = point.getNumber(1);
+                if (index == 0) {
+                    updateHoverText("Unassigned: " + unassignedTargets, x, y);
+                } else {
+                    updateHoverText(index + ": " + groupTargetCounts.get(index - 1), x, y);
+                }
+
+                return null;
+            }
+        }).on(BrowserEvents.MOUSEOUT, new DatumFunction<Void>() {
+            @Override
+            public Void apply(Element context, Value d, int index) {
+                infoText.attr(ATTR_VISIBILITY, "hidden");
+                return null;
+            }
+        });
+        pies.exit().remove();
+        pies.attr("d", arc);
+
+    }
+
+    private void updateHoverText(final String displayText, final double x, final double y) {
+        final Selection text = infoText.select("text");
+        final Selection background = infoText.select("rect");
+
+        text.html(displayText);
+
+        final double textWidth = getTextWidth(text.node());
+        final double textHeight = getTextHeight(text.node());
+
+        background.attr("width", textWidth * 1.1);
+        background.attr("height", textHeight);
+
+        moveSelection(background, -textWidth * 1.1 / 2.0, -textHeight*0.8);
+        moveSelection(infoText, x, y);
+        infoText.attr(ATTR_VISIBILITY, "visible");
+    }
+
+    private static void moveSelection(Selection sel, double x, double y) {
+        sel.attr(ATTR_TRANSFORM, "translate(" + x + ", " + y + ")");
+    }
+
+
+    private static final native double getTextWidth(Element e)/*-{
```

Comment:
The comment block is required, because it is the implementation of the native method. See [JSNI](http://www.gwtproject.org/doc/latest/DevGuideCodingBasicsJSNI.html).